File: leetcode/leetcode_27_remove_element/Solution.java
package leetcode_27_remove_element;

class Solution {
    public int removeElement(int[] nums, int val) {
        int slowIndex = 0;
        for (int i = 0; i < nums.length; i++ ){
            if (nums[i] != val){
                nums[slowIndex] = nums[i];
                slowIndex ++;
            }
        }
        return slowIndex;
    }
}

File: leetcode/leetcode_27_remove_element/solution.py
class Solution(object):
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        slow_index = 0
        fast_index = 0
        for fast_index in range(len(nums)):
            if nums[fast_index] != val:
                nums[slow_index] = nums[fast_index]
                slow_index += 1
        return slow_index        
        


s = Solution()
nums_test = [0, 1, 2, 2, 4, 5]
val_test = 2
s.removeElement(nums_test, val_test)
print(nums_test)
